Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -16,104 +16,86 @@ import UIKit
 import ARKit
 import ArcGIS
 
-extension ArcGISARView.CoreLocationError: CustomNSError {
-    static var errorDomain: String {
-        return kCLErrorDomain
-    }
-
-    var errorCode: Int {
-        return CLError.Code.denied.rawValue
-    }
-    
-    var errorDescription: String? {
-        switch self {
-        case .accessDenied:
-            return "Access to the device location is denied."
-        case .missingPListKey:
-            return "You must specify a location usage description key (NSLocationWhenInUseUsageDescription or NSLocationAlwaysUsageDescription) in your plist."
-        }
-    }
-}
-
 public class ArcGISARView: UIView {
-    
-    enum CoreLocationError: Swift.Error {
-        case accessDenied
-        case missingPListKey
-    }
 
     // MARK: public properties
     
     /// The view used to display the `ARKit` camera image and 3D `SceneKit` content.
     public let arSCNView = ARSCNView(frame: .zero)
     
-    /// The view used to display ArcGIS 3D content.
-    public let sceneView = AGSSceneView(frame: .zero)
+    /// The initial transformation used for a table top experience.  Defaults to the Identity Matrix.
+    public private(set) var initialTransformation = AGSTransformationMatrix.identity
     
-    /// The camera controller used to control the Scene
-    private let cameraController = AGSTransformationMatrixCameraController()
+    /// Denotes whether tracking location and angles has started.
+    public private(set) var isTracking: Bool = false
     
-    /// The world tracking information used by `ARKit`.
-    public var arConfiguration: ARConfiguration = {
-        let config = ARWorldTrackingConfiguration()
-        config.worldAlignment = .gravityAndHeading
-        return config
-        }() {
+    /// Denotes whether ARKit is being used to track location and angles.
+    public private(set) var isUsingARKit: Bool = true
+
+    /// The data source used to get device location.  Used either in conjuction with ARKit data or when ARKit is not present or not being used.
+    public var locationDataSource: AGSCLLocationDataSource? {
```

Comment:
I just want to confirm that this is intended to be publicly settable?

Reply:
Yes, publicly settable.  We allow clients to create and use their own LocationDataSources, like from a high-accuracy GPS or maybe Beacons.